### DISCUSSION TITLE
What is the eslint rule to to enforce a space after the closing parentheses of a single-line if statement?

### DISCUSSION THREAD

**Discussion by axefrog** [eslint/eslint#15267] [Q&A] [answered]
For the life of me I cannot find an ESLint rule to enforce the following:

```js
// Bad:
if (foo)bar;

// Good:
if (foo) bar;
```

Any assistance would be much appreciated!

**Comment by mdjermanovic** [comment]
I think we don't have that rule in the core.

`keyword-spacing` can disallow `if (foo)return;`, `space-before-blocks` can disallow `if (foo){`, but it seems that there is no rule that would disallow `if (foo)bar;` and enforce `if (foo) bar;`

**Reply by axefrog** [reply]
Ah, thanks for the answer. Single-line `if` statements are very common in JavaScript. Would you agree that this is something that should be there?

**Reply by axefrog** [reply]
@sketchbuch Care to comment on that thumbsdown reaction?

**Reply by axefrog** [reply]
I have unmarked this as an answer because just stating that it doesn't exist feels like an incomplete answer. Single line `if` statements are common in JavaScript. This feels like it should be in eslint core. Is its omission a design decision, and if so why? Or is this an oversight, and if so, could it be made a candidate for inclusion in the core in a future update?

**Comment by aladdin-add** [comment, marked as answer]
stylistic rules changes are frozen. refs: https://github.com/eslint/eslint#stylistic-rule-updates

You'll need to write your own.

**Reply by axefrog** [reply]
Thank you for clarifying.

**Reply by ftzi** [reply]
Is there a package that does this?

**Comment by jonlepage** [comment]
am also looking for this ! 
![image](https://user-images.githubusercontent.com/24865815/204846481-88cf68c6-6f3c-4ede-a59f-c79e96f142e2.png)
